python/ray/air/config.py

```diff
@@ -177,6 +177,19 @@ def __post_init__(self):
                     "`resources_per_worker."
                 )
 
+            # Validate inputs for resources_per_worker
+            accepted_resource_arguments: List[str] = ["GPU", "CPU", "memory", "custom"]
```

Comment:
I don't think this works fully because [custom resources](https://docs.ray.io/en/latest/ray-core/scheduling/resources.html#custom-resources) can be defined arbitrarily by the user.

Reply:
Ahh ok - is there still some way in that case to at least make it easier to detect silly errors like "memory" being written as "Memory"?

Just trying to save others in future from getting caught out and not knowing how to get to the bottom of the issue.

Reply:
I'm asking around to see if there's an existing API to fetch all possible resource keys available in the cluster - if there is one exposed then we can use that in place of this list!

Reply:
Hey Matthew,

Just following up on this - is there an existing API to perform this task?

Reply:
Hey unfortunately it does not exist yet - I've opened a feature request here https://github.com/ray-project/ray/issues/49501